Formula/glibc.rb

```diff
@@ -95,43 +88,191 @@ class Glibc < Formula
+    sha256 "9bba0214ccf7f1079c5d59210045227bcf619519840ebfa80cd3849cff5a5bf2"
+  end
 
+  resource "bzip2" do
+    url "https://sourceware.org/pub/bzip2/bzip2-1.0.8.tar.gz"
+    sha256 "ab5a03176ee106d3f0fa90e381da478ddae405918153cca248e682cd0c4a2269"
+  end
+
+  resource "gcc" do
+    url "https://ftp.gnu.org/gnu/gcc/gcc-10.4.0/gcc-10.4.0.tar.xz"
+    sha256 "c9297d5bcd7cb43f3dfc2fed5389e948c9312fd962ef6a4ce455cff963ebe4f1"
+  end
+
+  resource "mpfr" do
+    url "ftp://gcc.gnu.org/pub/gcc/infrastructure/mpfr-3.1.6.tar.bz2"
+    sha256 "cf4f4b2d80abb79e820e78c8077b6725bbbb4e8f41896783c899087be0e94068"
+  end
+
+  resource "mpc" do
+    url "ftp://gcc.gnu.org/pub/gcc/infrastructure/mpc-1.0.3.tar.gz"
+    sha256 "617decc6ea09889fb08ede330917a00b16809b8db88c29c31bfbb49cbf88ecc3"
+  end
+
+  resource "gmp" do
+    url "ftp://gcc.gnu.org/pub/gcc/infrastructure/gmp-6.1.0.tar.bz2"
+    sha256 "498449a994efeba527885c10405993427995d3f86b8768d8cdf8d9dd7c6b73e8"
+  end
+
+  def vendor_make?
+    if which("make").blank?
+      true
+    else
+      make_version = Utils.safe_popen_read("make", "--version").lines.first.split.last
+      Version.new(make_version) < MINIMUM_MAKE_VERSION
+    end
+  end
+
+  def install
     # Setting RPATH breaks glibc.
     %w[
       LDFLAGS LD_LIBRARY_PATH LD_RUN_PATH LIBRARY_PATH
       HOMEBREW_DYNAMIC_LINKER HOMEBREW_LIBRARY_PATHS HOMEBREW_RPATH_PATHS
     ].each { |x| ENV.delete x }
 
+    # TODO: make an audit exception or come up with a better way
+    # to use /usr/bin/gcc.
+    mkdir buildpath/"gcc/bin"
+    ln_sf "/usr/bin/gcc", buildpath/"gcc/bin"
+    ln_sf "/usr/bin/g++", buildpath/"gcc/bin"
+    ENV["CC"] = buildpath/"gcc/bin/gcc"
+    ENV["CXX"] = buildpath/"gcc/bin/g++"
+    ENV.prepend_path "PATH", buildpath/"bootstrap/bin"
+
+    if vendor_make?
+      resource("make").stage do
+        system "./configure", "--prefix=#{buildpath}/bootstrap"
+        system "make", "install"
+      end
+      ENV["MAKE"] = buildpath/"bootstrap/bin/make"
+    end
+
+    if which("python3").blank?
+      resource("python3").stage do
+        with_env(
+          CPPFLAGS: "-I#{Formula["zlib"].opt_include}",
+          LDFLAGS:  "-L#{Formula["zlib"].opt_lib} -Wl,-rpath,#{Formula["zlib"].opt_lib}",
+        ) do
+          system "./configure", "--prefix=#{buildpath}/bootstrap"
+          system "make"
+          system "make", "install"
+        end
+      end
+    end
+
+    if which("bison").blank?
+      args = ["--prefix=#{buildpath}/bootstrap"]
+      if which("m4").blank?
+        resource("m4").stage do
+          system "./configure", *args
+          system "make"
+          system "make", "install"
+        end
+        args << "M4=#{buildpath}/bootstrap/bin/m4"
+      end
+
+      resource("bison").stage do
+        system "./configure", *args
+        system "make", "install"
+      end
+    end
+
+    gcc_version = Utils.safe_popen_read("gcc", "--version").lines.first.split[2]
```

Comment:
Isn't it easier to use `gcc -dumpversion` here?

Reply:
Yes, I didn't know that existed.  Sadly no such equivalent exists for `make` but at least we can use it for `gcc`.